docs/Standard-RESTful-API.md

```diff
@@ -0,0 +1,520 @@
+
+Each PowerAuth implementation that is located on a specific base URL then has `/pa/` prefixed endpoints by convention.
+
+Following endpoints are published in PowerAuth Standard RESTful API (protocol version 3):
+
+- [`/pa/v3/activation/create`](#initiate-activation) - Create a new activation, perform a key exchange based on activation code.
+- [`/pa/v3/activation/status`](#activation-status) - Query for an activation status.
+- [`/pa/v3/activation/remove`](#activation-remove) - Remove an activation (requires authentication).
+- [`/pa/v3/token/create`](#create-token) - Create a token (requires authentication and encryption).
+- [`/pa/v3/token/remove`](#remove-token) - Remove a token (requires authentication).
+- [`/pa/v3/vault/unlock`](#vault-unlock) - Get a key to unlock secure vault (requires authentication and encryption).
+- [`/pa/v3/signature/validate`](#validate-signature) - Validate a signature (requires authentication).
+- [`/pa/v3/upgrade/start`](#upgrade-start) - Start a protocol upgrade (requires encryption).
+- [`/pa/v3/upgrade/commit`](#upgrade-commit) - Commits a protocol upgrade (requires authentication).
+
+Before you continue, you can also read [End-To-End encryption](./End-To-End-Encryption.md) and [Computing and Validating Signatures](./Computing-and-Validating-Signatures.md) documents, describing encryption and authentication, used in the RESTful API.
+
+## Error handling
+
+PowerAuth Standard RESTful API uses a unified format for error response body, accompanied with an appropriate HTTP status code. Besides the HTTP error codes that application server may return regardless of server application (such as 404 when resource is not found or 503 when server is down), following status codes may be returned:
+
+| Code | Description |
+|------|-------------|
+| 200  | OK response, no issues |
+| 400  | Issue with a request format, or issue of the business logic |
+| 401  | Unauthorized, signature validation failed for authenticated endpoints |
+
+All error responses that are produced by the PowerAuth Standard RESTful API have following body:
+
+```json
+{
+    "status": "ERROR",
+    "responseObject": {
+        "code": "ERROR_CODE",
+        "message": "ERROR_MESSAGE_IN_ENGLISH"
+    }
+}
+```
+
+Note that in case of PowerAuth, the code and message is usually very generic and does not provide a lot of information. Please consult the server log for details. On application level, use HTTP status codes to determine the type of the issue and present appropriate message to the user.
+
+## Initiate activation
```

Comment:
The chapter name is confusing. We use `initActivation` SOAP method to initiate activation on the server in the beginning of the activation process. So the chapter name should use the word `create` or something similar, but definitely not `initiate` which indicates association with `initActivation` SOAP method.

Reply:
SOAP method doesn't always match RESTful API. The title is the same as was for 2.0 version of protocol.